### ISSUE TITLE
Add support for configuration files describing formatter style

### ISSUE THREAD

**Issue by mglb** [chipsalliance/verible#898]
The formatting style is currently controlled using command line switches:

```
-- | assignment_statement   | _alignment
   | case_items             |
   | class_member_variables |
   | net_variable           |
   | struct_union_members   |
   | formal_parameters      |
   | named_parameter        |
   | named_port             |
   | port_declarations      |

-- | formal_parameters | _indentation
   | named_parameter   |
   | named_port        |
   | port_declarations |
```

https://github.com/chipsalliance/verible/pull/877 adds:

```
-- | port_declarations  | _right_align | _packed   | _dimensions
   | (...more someday?) |              | _unpacked |
```

There could (will?) be a lot more - in many cases configurability is trivial to implement.

I think it would be useful (both for Verible devs and users) to move style configuration to a configuration file. Preferably something with a tree structure and wildcards, so that something like this would be possible:

```
*.alignment = align
case_items.alignment = flush-left

*.dimensions.*.alignment = left
```

Verible linter already supports config files, so we could just extend its format: https://github.com/chipsalliance/verible/blob/master/common/text/config_utils.cc (short format description: https://github.com/chipsalliance/verible/tree/master/verilog/tools/lint#rule-configuration )

Additional features that could be useful later:
- Support for sections/mutliple configs in one file - it would allow to use single configuration file for Verible linter and formatter.

**Comment by bensampson5** [comment]
I would like to second the importance of this. Currently, I wrote a workaround in Python for the formatter that looks for a [.verible-verilog-format.yaml](https://github.com/bensampson5/libsv/blob/main/.verible-verilog-format.yaml) in the current working directory, pulls out all the options, and passes it to ``verible-verilog-format``. It would be nice if this support automated source file discovery and config files.

**Comment by mithro** [comment]
I think @hzeller was working on something like this?

**Comment by kaddkaka** [comment]
> I would like to second the importance of this. Currently, I wrote a workaround in Python for the formatter that looks for a [.verible-verilog-format.yaml](https://github.com/bensampson5/libsv/blob/main/.verible-verilog-format.yaml) in the current working directory, pulls out all the options, and passes it to `verible-verilog-format`. It would be nice if this support automated source file discovery and config files. So essentially all you would need to do would be run `verible-verilog-format .` and it would take care of the rest

Automatic pickup of config would be great! 😁 

As long as there are options to
- print information about the config used with debug or verbose flag  `-v`
- `--isolated` to ignore config file and only use settings from command line
- explicitly select config file and ignore the auto pickup.

**Comment by hzeller** [comment]
Yeah, we need a configuration file. I would avoid yaml as it is a can of worms, but something that is derived from the names of the flags would be good (similar to `.clang-format`).
I remember I did something in that direction at some point, but I might need to reactivate that.
All settings are in `./verible/verilog/formatting/format-style-init.cc` so that would be a good place to start.
We also have a configuration for the linter, so maybe there is some code to share.
